ecosystem/typescript/sdk_v2/src/transactions/instances/transactionPayload.ts

```diff
@@ -134,21 +136,52 @@ export class EntryFunction {
    * public entry fun transfer<CoinType>(from: &signer, to: address, amount: u64)
    * ```
    */
-  constructor(module_name: ModuleId, function_name: Identifier, type_args: Array<TypeTag>, args: Array<Uint8Array>) {
+  constructor(module_name: ModuleId, function_name: Identifier, type_args: Array<TypeTag>, args: Array<Serializable>) {
     this.module_name = module_name;
     this.function_name = function_name;
     this.type_args = type_args;
     this.args = args;
   }
 
+  /**
+   * A helper function to build a EntryFunction payload from raw primitive values
+   *
+   * @param module_name Fully qualified module name in format "AccountAddress::module_name" e.g. "0x1::coin"
+   * @param function_name Function name
+   * @param type_args Type arguments that move function requires.
+   *
+   * @example
+   * A coin transfer function has one type argument "CoinType".
+   * ```
+   * public(script) fun transfer<CoinType>(from: &signer, to: address, amount: u64,)
+   * ```
+   * @param args Arugments to the move function.
+   *
+   * @example
+   * A coin transfer function has three arugments "from", "to" and "amount".
+   * ```
+   * public(script) fun transfer<CoinType>(from: &signer, to: address, amount: u64,)
+   * ```
+   * @returns EntryFunction
+   */
+  static build(
+    module_name: `${string}::${string}`,
+    function_name: string,
+    type_args: Array<TypeTag>,
+    args: Array<Serializable>,
+  ): EntryFunction {
+    return new EntryFunction(ModuleId.fromStr(module_name), new Identifier(function_name), type_args, args);
+  }
+
   serialize(serializer: Serializer): void {
     this.module_name.serialize(serializer);
     this.function_name.serialize(serializer);
     serializer.serializeVector<TypeTag>(this.type_args);
 
     serializer.serializeU32AsUleb128(this.args.length);
-    this.args.forEach((item: Uint8Array) => {
-      serializer.serializeBytes(item);
+    this.args.forEach((item: Serializable) => {
+      const bytes = item.bcsToBytes();
+      serializer.serializeBytes(bytes);
```

Comment:
So BTW, I realized this really sucks in terms of consistent representations, because it means whenever we deserialize a transaction, we have to treat the bytes as the FixedBytes thing.

I'll add more info about this later, I have a solution (I think) but it changes a lot of the interfaces for stuff, although generally I think this is sorely needed, because `Serializable` isn't really an accurate representation of what we're sending in here